diagnostic_updater/diagnostic_updater/_diagnostic_updater.py

```diff
@@ -206,90 +217,98 @@ def removeByName(self, name):
         return found
 
 
-
-
 class Updater(DiagnosticTaskVector):
-    """Manages a list of diagnostic tasks, and calls them in a rate-limited manner.
+    """
+    Manage a list of diagnostic tasks, and calls them in a rate-limited manner.
 
     This class manages a list of diagnostic tasks. Its update function
     should be called frequently. At some predetermined rate, the update
     function will cause all the diagnostic tasks to run, and will collate
     and publish the resulting diagnostics. The publication rate is
     determined by the "~diagnostic_period" ros parameter.
-
     The class also allows an update to be forced when something significant
     has happened, and allows a single message to be broadcast on all the
     diagnostics if normal operation of the node is suspended for some
     reason.
     """
 
-    def __init__(self):
-        """Constructs an updater class."""
+    def __init__(self, node):
+        """Construct an updater class."""
         DiagnosticTaskVector.__init__(self)
-        self.publisher = rospy.Publisher("/diagnostics", DiagnosticArray, queue_size=10)
+        self.node = node
+        self.publisher = self.node.create_publisher(DiagnosticArray, '/diagnostics')
+        clock = Clock(clock_type=ClockType.STEADY_TIME)
```

Comment:
Instead of constructing a `Clock` every time one is needed in this class, does it make more sense to create it once in the constructor and store it in a member variable?